workflow/controller/operator.go

```diff
@@ -1151,7 +1151,11 @@ func printPodSpecLog(pod *apiv1.Pod, wfName string) {
 // and returns the new node status if something changed
 func (woc *wfOperationCtx) assessNodeStatus(pod *apiv1.Pod, old *wfv1.NodeStatus) *wfv1.NodeStatus {
 	new := old.DeepCopy()
-	tmpl := woc.GetNodeTemplate(old)
+	tmpl, err := woc.GetNodeTemplate(old)
```

Comment:
do we need to check the tmpl is nil or not?

Reply:
was debating that one. err should be non-nil in that case, but I could add an additional check for safety. If tmpl is nil and err is non-nil, then that would be a bad unexpected situation. Would we rather crash and maybe be alerted that this case occurred through the stack trace or log an error (not sure if it will be seen?)

Reply:
I say alert over the crash, don't want to be in a situation where 1 workflow crashes the controller causing all other workflows to hang

Reply:
Actually, I realized (based on an e2e test) that it seems to be valid for tmpl to be nil. For safety I added a couple of checks in the switch cases below before we try to use it to make sure it's not nil.